tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -1001,70 +1085,111 @@ def convolution(inputs,
     ValueError: If `data_format` is invalid.
     ValueError: Both 'rate' and `stride` are not uniformly 1.
   """
-  if data_format not in [None, 'NWC', 'NCW', 'NHWC', 'NCHW', 'NDHWC', 'NCDHW']:
-    raise ValueError('Invalid data_format: %r' % (data_format,))
+  return _convolution(inputs,
```

Comment:
Why did you create _convolution instead of just using convolution?

Reply:
@sguada Thanks for the review. The `_convolution` has one additional args of `dims=None` (compared to `convolution`) and has the following logic added:
https://github.com/yongtang/tensorflow/blob/bc6128be953fa4163743dad63686d64fdfe720b8/tensorflow/contrib/layers/python/layers/layers.py#L948-L950

In this way, `convolution` will call `_convolution` with `dims=None`, `convolution2d` will call `_convolution` with `dims=2`, and `convolution2d` will call `_convolution` with `dims=3`.

The idea is to leave the api of `convolution/convolution2d/convolution3d` intact, while at the same time only check dims on `convolution2d/convolution3d`

Reply:
Can you add an extra key argument to convolution with default that doesn't change current behavior?

Reply:
Thanks @sguada. The PR has been updated. Please take a look.